app/nagbot.py

```diff
@@ -139,67 +101,45 @@ def notify(self, channel, dryrun):
-            for i in instances_to_terminate:
-                contact = sqslack.lookup_user_by_email(i.contact)
-                message = message + sqaws.Instance.make_resource_summary(blank_instance, i) \
-                    + ', "Terminate after"={}, "Monthly Price"={}, Contact={}\n' \
-                    .format(i.terminate_after, money_to_string(i.monthly_price), contact)
-                sqaws.Instance.terminate_resource(i, i.region_name, i.instance_id, dryrun=dryrun)
-            sqslack.send_message(channel, message)
-        else:
-            sqslack.send_message(channel, 'No instances were terminated today.')
-
-        if len(volumes_to_delete) > 0:
-            message = 'I deleted the following volumes: '
-            for v in volumes_to_delete:
-                contact = sqslack.lookup_user_by_email(v.contact)
-                message = message + sqaws.Volume.make_resource_summary(blank_volume, v) \
-                    + ', "Terminate after"={}, "Monthly Price"={}, Contact={}\n' \
-                    .format(v.terminate_after, money_to_string(v.monthly_price), contact)
-                sqaws.Volume.terminate_resource(v, v.region_name, v.volume_id, dryrun=dryrun)
-            sqslack.send_message(channel, message)
-        else:
-            sqslack.send_message(channel, 'No volumes were deleted today.')
-
-        if len(instances_to_stop) > 0:
-            message = 'I stopped the following instances: '
-            for i in instances_to_stop:
-                contact = sqslack.lookup_user_by_email(i.contact)
-                message = message + sqaws.Instance.make_resource_summary(blank_instance, i) \
-                    + ', "Stop after"={}, "Monthly Price"={}, Contact={}\n' \
-                    .format(i.stop_after, money_to_string(i.monthly_price), contact)
-                sqaws.Instance.stop_instance(i.region_name, i.instance_id, dryrun=dryrun)
-                sqaws.set_tag(i.region_name, 'instance', i.instance_id, i.nagbot_state_tag_name, 'Stopped on ' +
-                              TODAY_YYYY_MM_DD, dryrun=dryrun)
-            sqslack.send_message(channel, message)
-        else:
-            sqslack.send_message(channel, 'No instances were stopped today.')
+        resource_types = [Instance, Volume]
+        for resource_type in resource_types:
+            ec2_type, ec2_state = resource_type.to_string()
+            resources = resource_type.list_resources()
+
+            # Only terminate resources which still meet the criteria for terminating, AND were warned several times
+            resources_to_terminate = list(r for r in resources if r.is_terminatable(r, TODAY_YYYY_MM_DD) and
+                                          r.is_safe_to_terminate(r, TODAY_YYYY_MM_DD))
+
+            # Only stop resources which still meet the criteria for stopping, AND were warned recently
+            resources_to_stop = list(r for r in resources if sqaws.is_stoppable(r, TODAY_YYYY_MM_DD) and
```

Comment:
Could `is_stoppable` and `is_safe_to_stop` here move to be methods on the resource?